### PR TITLE
Adding menus documentation.

### PR DESCRIPTION
**Pull request by cdriscol**
Related to #743, also assumes #778 gets merged.

### CODE REVIEW

**Comment by lirantal** [comment]
Great, thanks Chris!

File: _includes/docs/0.4.x/angular/menus.html [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  function(Menus) {
+    Menus.addMenuItem('topbar', {title: 'Example', state: 'example'});
+    
+    Menus.addMenuItem('topbar', {title: 'Example Dropdown', state: 'exampledropdown', type: 'dropdown'});
+    Menus.addSubMenuItem('topbar', 'exampledropdown', {title: 'Example Sub Item', state: 'examplesubitem'});
+  }
+]);
+</pre>
+  <p class="alert alert-info">
+    To learn more about the Menus service, visit <a href="#article-example">The Article Example Section</a>.
```
Comment (codydaig) [inline _includes/docs/0.4.x/angular/menus.html]: i think this line should be commented out until we get the article example up.

**Comment by codydaig** [comment]
@cdriscol If you can comment out the the reference to the article example for now and remove references to the isPublic option since that has been removed, this will be ready to merge. :-D Thanks!

**Comment by ilanbiala** [comment]
@cdriscol @codydaig any update on this?

**Comment by codydaig** [comment]
@ilanbiala Just waiting on @cdriscol to fix the couple small things and this can be merged.

**Comment by mg1075** [comment]
The menu documentation leaves me with more questions than answers and feels like it needs to be considerably expanded upon. I asked a question about menus in gitter with no feedback given. 

http://meanjs.org/docs/0.4.x/

What *is* a menu object?

What is the difference between a "type" of 'dropdown' and of 'item'?

What is the significance of the **topbar** menu? Does declaring "topbar" magically put the menu in a top navbar?

What if I want to put a an unordered list of "menu" items on the left or right side of a page and I do not want to use a dropdown? How does the documentation help me understand how this task should be done?

**Comment by codydaig** [comment]
@mg1075 Can you post those questions to a new issue so it doesn't get lost in this closed PR? We can attempt to address these concerns.

**Comment by mg1075** [comment]
@codydaig - done. 
https://github.com/meanjs/mean/issues/1105

**Pull request closed without merging**